proto/proto/block_producer.proto

```diff
@@ -2,22 +2,18 @@
 syntax = "proto3";
 package block_producer;
 
-import "requests.proto";
-import "responses.proto";
-
-import "block.proto";
-import "digest.proto";
-import "note.proto";
-import "transaction.proto";
-
+import "types/account.proto";
+import "types/blockchain.proto";
+import "types/primitives.proto";
+import "types/shared.proto";
 import "google/protobuf/empty.proto";
 
 service Api {
     // Submits proven transaction to the Miden network
-    rpc SubmitProvenTransaction(requests.SubmitProvenTransactionRequest) returns (responses.SubmitProvenTransactionResponse) {}
+    rpc SubmitProvenTransaction(shared.SubmitProvenTransaction) returns (shared.ProvenTransaction) {}
 
     // Returns the status info.
-    rpc Status(google.protobuf.Empty) returns (responses.BlockProducerStatusResponse) {}
+    rpc Status(google.protobuf.Empty) returns (shared.BlockProducerStatus) {}
```

Comment:
Could we define `BlockProducerStatus` in this file? I believe it needs to be shared with the `rpc.proto`, but we should be able to import `block_producer.proto` there, no?